src/arrayMethodJoin.js

```diff
@@ -5,7 +5,35 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let joinedStr = '';
+    let separatorInn = separator;
+    const arrInn = [...this];
+
+    if (separator === undefined) {
+      separatorInn = ',';
+    }
```

Comment:
add default separator like this instead of if block
[].__proto__.join2 = function(separator = ',')